### ISSUE TITLE
Changing Style in the layer control should not close it

### ISSUE THREAD

**Issue by Malvoz** [Maps4HTML/Web-Map-Custom-Element#462]
Repro steps:

1. Visit https://maps4html.org/experiments/datacube/
2. Open the layer control and click a radio input to change Style of any layer

The result is that the layer control closes and focus moves away from the component. This is a failure of WCAG 2.1 [SC 3.2.2 On Input](https://www.w3.org/TR/WCAG21/#on-input).

**Comment by prushforth** [comment]
This is going to be hard. The 'style' is actually a link to an alternate layer. When the user selects the style option, the other layer is loaded, replacing the entire contents of the current layer, including the entry in the layer control. The incoming layer may or may not 'know' about the current layer, meaning even the style radio group may be removed and not replaced.

**Comment by Malvoz** [comment]
Okay, not sure how to best deal with that. But IMO this is low priority compared to other accessibility issues.

**Comment by prushforth** [comment]
Yes, agreed. The UI as implemented leads one to think it's just a choice, but in fact it's a link similar in behaviour to `<map-a target= "_self">`.

**Comment by Malvoz** [comment]
I think the most logical thing right now would be for the radio inputs to be part of a `<form>` and an Update button to trigger the change, something like:

![style-form](https://user-images.githubusercontent.com/26493779/117841974-ca529d00-b27d-11eb-8a03-696c82632d3d.png)

**Comment by prushforth** [comment]
I checked Google Maps and it uses a radio menu option group for the same function, also closing the layer menu after selection. Not saying they're perfect but they screwed up in the same way at least.

**Comment by Malvoz** [comment]
Okay, I could report that in https://issuetracker.google.com/issues/69541792.

**Comment by Malvoz** [comment]
> I checked Google Maps and it uses a radio menu option group for the same function, also closing the layer menu after selection.

Where did you check that? The [Google Maps Embed](https://maps4html.org/HTML-Map-Element-UseCases-Requirements/examples/create-map.html#google-maps-embed) has a control to switch between <q>street map</q> and <q>satellite imagery</q> which is a simple toggle button (not keyboard operable however). And the [Google Maps site](https://www.google.com/maps/) has [`menuitemcheckbox`](https://www.w3.org/TR/wai-aria/#menuitemcheckbox) controls for changing layers, and the layer menu does not close and layer items do not lose focus after changing the settings.

**Comment by Malvoz** [comment]
Side note: Is there a particular reason why in https://maps4html.org/experiments/datacube/ the first layer in the list doesn't have a Style pre-selected (checked)? I don't think there's a way for keyboard users to reach them.

**Comment by prushforth** [comment]
> Side note: Is there a particular reason why in https://maps4html.org/experiments/datacube/ the first layer in the list doesn't have a Style pre-selected (checked)? I don't think there's a way for keyboard users to reach them.

It's a bug, afaict.  

> Google Maps has improved keyboard accessibility quite a bit in general since we reported issues last year, so maybe their layer controls have changed since you tested them?

I believe that's true, I swear it was an option group when I made this comment:

> > I checked Google Maps and it uses a radio menu option group for the same function, also closing the layer menu after selection.

But today it is indeed a button as you say:

![image](https://user-images.githubusercontent.com/2437285/128934942-5de7fa23-4d50-4ac9-ba1f-95100ce3788a.png)

The location of the layers selector has now moved, so I presume this change was made when they moved it:

![image](https://user-images.githubusercontent.com/2437285/128935239-3b6e5320-6e87-404d-b251-c7d0ad710e94.png)

![image](https://user-images.githubusercontent.com/2437285/128935179-0d94d414-7716-474d-a6b2-1ce062bec787.png)
